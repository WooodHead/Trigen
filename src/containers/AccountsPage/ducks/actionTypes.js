export const ADD_ACCOUNT = 'ADD_ACCOUNT';
export const FETCH_ACCOUNTS = 'FETCH_ACCOUNTS';
export const FETCH_ACCOUNTS_SUCCESS = 'FETCH_ACCOUNTS_SUCCESS';
export const FETCH_ACCOUNTS_ERROR = 'FETCH_ACCOUNTS_ERROR';
export const EDIT_ACCOUNT = 'EDIT_ACCOUNT';
export const UPDATE_ACCOUNT = 'UPDATE_ACCOUNT';
export const UPDATE_ACCOUNT_SUCCESS = 'UPDATE_ACCOUNT_SUCCESS';
export const UPDATE_ACCOUNT_ERROR = 'UPDATE_ACCOUNT_ERROR';
export const SAVE_NEW_ACCOUNT = 'SAVE_NEW_ACCOUNT';
export const SAVE_NEW_ACCOUNT_SUCCESS = 'SAVE_NEW_ACCOUNT_SUCCESS';
export const SAVE_NEW_ACCOUNT_ERROR = 'SAVE_NEW_ACCOUNT_ERROR';
export const CANCEL_EDIT = 'CANCEL_EDIT';
export const CHANGE_ACCOUNT = 'CHANGE_ACCOUNT';
export const DELETE_ACCOUNT = 'DELETE_ACCOUNT';
export const DELETE_ACCOUNT_SUCCESS = 'DELETE_ACCOUNT_SUCCESS';
export const DELETE_ACCOUNT_ERROR = 'DELETE_ACCOUNT_ERROR';
export const FILTER_ACCOUNT = 'FILTER_ACCOUNT';
