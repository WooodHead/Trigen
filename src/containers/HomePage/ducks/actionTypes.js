export const GENERATE_PASSWORD = 'GENERATE_PASSWORD';
export const GENERATE_PASSWORD_SUCCESS = 'GENERATE_PASSWORD_SUCCESS';
export const GENERATE_PASSWORD_ERROR = 'GENERATE_PASSWORD_ERROR';
export const CLEAR_PASSWORD = 'CLEAR_PASSWORD';
