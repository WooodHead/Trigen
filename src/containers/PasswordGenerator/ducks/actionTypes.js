export const GENERATE_PASSWORD = 'GENERATE_PASSWORD';
export const GENERATE_PASSWORD_SUCCESS = 'GENERATE_PASSWORD_SUCCESS';
export const GENERATE_PASSWORD_ERROR = 'GENERATE_PASSWORD_ERROR';
export const CLEAR_PASSWORD = 'CLEAR_PASSWORD';
export const START_PROGRESS = 'START_PROGRESS';
export const STOP_PROGRESS = 'STOP_PROGRESS';
