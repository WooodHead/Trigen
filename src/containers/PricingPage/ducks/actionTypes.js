export const BUY_SLOTS = 'BUY_SLOTS';
export const BUY_SLOTS_SUCCESS = 'BUY_SLOTS_SUCCESS';
export const BUY_SLOTS_ERROR = 'BUY_SLOTS_ERROR';
