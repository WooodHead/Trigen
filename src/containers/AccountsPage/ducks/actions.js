import * as actionTypes from './actionTypes';

// Actions

/**
  * Adds a new account entry to the accounts list and to the EditedAccounts list.
  *
  * @param {string} fakeID -- Internal ID to refer to and synchronize entries in the accounts
  * and EditedAccounts lists
  *
  * @return {Object} -- An action object with the type ADD_ACCOUNT
  */
export function addAccount(fakeID) {
  return {
    type: actionTypes.ADD_ACCOUNT,
    payload: { data: {
      username: '',
      domain: '',
      new: true,
      edit: true,
      fakeID,
    } }
  };
}

export function fetchAccounts(userID, authToken, genIDFunc) {
  return {
    type: actionTypes.FETCH_ACCOUNTS,
    payload: { userID, authToken, genIDFunc }
  };
}

/**
  * Changes the data of an entry inside the EditedAccount list.
  *
  * @param {String} fakeID -- Internal identifier to synchronize entries between lists
  * @param {Object} data -- New data for the updated entry
  *
  * @return {Object} -- Action with the type EDIT_ACCOUNT
  */
export function editAccount(fakeID, data) {
  return {
    type: actionTypes.EDIT_ACCOUNT,
    payload: { fakeID, data: {
      ...data, edit: true
    } },
  };
}

/**
 * Transfers data from an EditedAccount to the respective account inside
 * the accounts list.
 * TODO
 * A request needs to be implemented which patches/posts the new data to the server
  */
export function saveNewAccount(userID, authToken, EditedAccount) {
  return {
    type: actionTypes.SAVE_NEW_ACCOUNT,
    payload: { userID, authToken, EditedAccount },
  };
}

// TODO save existing account

/**
 * Deletes an EditedAccount from the respective list and basically reverts all changes
 * made.
 *
 * @param {String} fakeID -- Internal entry reference
 *
 * @return {Object} -- Action of the type CANCEL_EDIT
  */
export function cancelEdit(fakeID) {
  return {
    type: actionTypes.CANCEL_EDIT,
    payload: { fakeID }
  };
}

/**
  * Deletes an account from the accounts list.
  * TODO
  * A request needs to be implemented which also deletes the account from the database
  *
  * @param {String} fakeID -- Internal entry reference
  *
  * @return {Object} -- Aciton of the type DELETE_ACCOUNT
  */
export function deleteAccount(fakeID) {
  return {
    type: actionTypes.DELETE_ACCOUNT,
    payload: { fakeID }
  };
}

/**
 * Transfers the user made changes on an input to the respective entry inside
 * the EditedAccounts list.
 *
 * @param {String} fakeID -- Internal reference to the respective entry
 * @param {Object} target -- Target of the event which should contain a name and a value as properties
 *
 * @return {Object} -- Action with the type CHANGE_ACCOUNT
  */
export function handleAccountChange(fakeID, target) {
  return {
    type: actionTypes.CHANGE_ACCOUNT,
    payload: { fakeID, target }
  };
}

/**
  * Applies a filter to the account list so only certain accounts are shown
  *
  * @param {String} filter -- Filter expression to search for inside usernames and domains
  *
  * @return {Object} -- Action with type FILTER_ACCOUNT
  */
export function filterAccounts(filter) {
  return {
    type: actionTypes.FILTER_ACCOUNT,
    payload: { filter }
  };
}
