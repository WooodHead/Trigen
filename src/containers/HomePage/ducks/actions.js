import * as types from './actionTypes';

export const generatePassword = (accountInformation, userInfo, masterPassword) => {
  return {
    type: types.GENERATE_PASSWORD,
    payload: { accountInformation, userInfo, masterPassword }
  };
};
