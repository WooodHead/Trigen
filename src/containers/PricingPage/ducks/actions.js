import * as types from './actionTypes';

export const purchaseSlots = (userInfo, amount) => {
  return {
    type: types.BUY_SLOTS,
    payload: { userInfo, amount }
  };
};
