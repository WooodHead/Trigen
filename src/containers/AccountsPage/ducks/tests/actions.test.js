import * as types from '../actionTypes';
import * as actions from '../actions';


describe('actions', () => {
  it('should create an action to add an account', () => {
    const expectedAction = {
      type: types.ADD_ACCOUNT,
      payload: { data: {
        username: '',
        domain: '',
        new: true,
        edit: true,
        fakeID: 'fakeID',
      }}
    };

    expect(actions.addAccount('fakeID')).toEqual(expectedAction);
  });

  it('should create an action to fetch accounts', () => {
    const expectedAction = {
      type: types.GET_ACCOUNTS,
      payload: {
        accounts: [
          {
            id: 0,
            username: 'tTester',
            domain: 'example.com'
          },
          {
            id: 1,
            username: 'LSP',
            domain: 'Ooo.com'
          }
        ],
        genID: 'insert function here',
      }
    };

    expect(actions.fetchAccounts('insert function here')).toEqual(expectedAction);
  });

  it('should create an action to set account to edit mode', () => {
    const expectedAction = {
      type: types.EDIT_ACCOUNT,
      payload: { fakeID: 'fakeID', data: {
        test: 'test',
        edit: true,
      }}
    };

    expect(actions.editAccount('fakeID', {
      test: 'test',
    })).toEqual(expectedAction);
  });

  it('should create an action to save an edited account', () => {
    const expectedAction = {
      type: types.SAVE_ACCOUNT,
      payload: {
        test: 'test',
      }
    };

    expect(actions.saveAccount({ test: 'test' })).toEqual(expectedAction);
  });

  it('should create an action to cancel edited changes', () => {
    const expectedAction = {
      type: types.CANCEL_EDIT,
      payload: { fakeID: 'fakeID' }
    };

    expect(actions.cancelEdit('fakeID')).toEqual(expectedAction);
  });

  it('should create an action to delete an account', () => {
    const expectedAction = {
      type: types.DELETE_ACCOUNT,
      payload: { fakeID: 'fakeID' }
    };

    expect(actions.deleteAccount('fakeID')).toEqual(expectedAction);
  });

  it('should create an action to handle account changes', () => {
    const expectedAction = {
      type: types.CHANGE_ACCOUNT,
      payload: { fakeID: 'fakeID', target: {
        test: 'test'
      }}
    };

    expect(actions.handleAccountChange('fakeID', { test: 'test' })).toEqual(expectedAction);
  });
});
