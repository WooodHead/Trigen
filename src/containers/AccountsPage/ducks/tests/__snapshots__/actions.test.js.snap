exports[`actions should create an action to add an account 1`] = `
Object {
  "payload": Object {
    "data": Object {
      "domain": "",
      "edit": true,
      "fakeID": "fakeID",
      "new": true,
      "username": "",
    },
  },
  "type": "ADD_ACCOUNT",
}
`;

exports[`actions should create an action to cancel edited changes 1`] = `
Object {
  "payload": Object {
    "fakeID": "fakeID",
  },
  "type": "CANCEL_EDIT",
}
`;

exports[`actions should create an action to delete an account 1`] = `
Object {
  "payload": "uID",
  "type": "DELETE_ACCOUNT",
}
`;

exports[`actions should create an action to fetch accounts 1`] = `
Object {
  "payload": Object {
    "authToken": "tokenXYZ",
    "genIDFunc": "insert function here",
    "userID": "userID",
  },
  "type": "FETCH_ACCOUNTS",
}
`;

exports[`actions should create an action to handle account changes 1`] = `
Object {
  "payload": Object {
    "fakeID": "fakeID",
    "target": Object {
      "test": "test",
    },
  },
  "type": "CHANGE_ACCOUNT",
}
`;

exports[`actions should create an action to save a new account 1`] = `
Object {
  "payload": Object {
    "EditedAccount": undefined,
    "authToken": undefined,
    "userID": Object {
      "test": "test",
    },
  },
  "type": "SAVE_NEW_ACCOUNT",
}
`;

exports[`actions should create an action to set account to edit mode 1`] = `
Object {
  "payload": Object {
    "data": Object {
      "edit": true,
      "test": "test",
    },
    "fakeID": "fakeID",
  },
  "type": "EDIT_ACCOUNT",
}
`;
